src/lib/components/Textfield.react.js

```diff
@@ -0,0 +1,139 @@
+
+type Props = {
+  /** If true, the input element will be focused during the first mount */
+  autoFocus?: boolean,
+  /** The classes to be applied to this component. This keys in this object must be valid CSS rule
+   * names, and the values must be strings for the classnames to be assigned to each rule name
+   * Valid rule names are:
+    root
+    formControl
+    focused
+    disabled
+    colorSecondary
+    underline
+    error
+    multiline
+    fullWidth
+    input
+    inputMultiline
+   */
+  classes?: {
+    root?: string,
+    formControl?: string,
+    focused?: string,
+    disabled?: string,
+    colorSecondary?: string,
+    underline?: string,
+    error?: string,
+    multiline?: string,
+    fullWidth?: string,
+    input?: string,
+    inputMultiline?: string,
+  },
+  /** CSS class name of the root element */
+  className?: string,
+  /** 'primary' | 'secondary' */
+  color: string,
+  /** If true, the input element will be disabled. */
+  disabled?: boolean,
+  /** If true, the label will be displayed in an error state */
+  error?: boolean,
+  /** Dash callback to trigger an event handler */
+  fireEvent?: () => void,
+  /** If true, the input will take up the full width of its container */
+  fullWidth?: boolean,
+  /** The helper text content */
+  helperText?: Node,
+  /** Element ID */
+  id?: string,
+  /** If true, a textarea element will be rendered instead of an input */
+  multiline?: boolean,
+  /** If true, the label is displayed as required and the input element` will be required */
+  required?: boolean,
+  /** Dash callback to update props on the server. */
+  setProps?: () => void,
+  /** 'medium' | 'small' */
+  size?: string,
+  /** Override the inline-styles of the root element */
+  style?: Object,
+  /** Type of the input element. It should be a valid HTML5 input type */
+  type?: string,
+  /** The value of the input element, required for a controlled component */
+  value?: any,
+  /** 'filled' | 'outlined' | 'standard' */
+  variant?: string,
+};
+
+type State = {
+  disabled: boolean,
+  value: any,
+};
+
+const defaultProps = {
+  autoFocus: false,
+  classes: {},
+  className: '',
+  color: 'primary',
+  disabled: false,
+  error: false,
+  fireEvent: () => {},
+  fullWidth: false,
+  helperText: '',
+  multiline: false,
+  required: false,
+  setProps: () => {},
+  size: 'medium',
+  style: {},
+  type: 'text',
+  value: '',
+  variant: 'standard'
+};
+
+/** Material UI TextField component */
+export default class TextField extends Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      disabled: this.props.disabled,
+      value: this.props.value,
+    };
+    /** _.debounce used to provide delay in callback to avoid firing callback every
+     * time user input changes - waits this.props.dashCallbackDelay ms to fire callback */
+    this.updateTextProps = _.debounce(this._updateTextProps, this.props.dashCallbackDelay);
```

Comment:
Good idea to use debounce for this component! This will require a few more pieces, such as adding another prop for `dashCallbackDelay` and further defining `updateTextProps`. You can reference [this component](https://github.com/StratoDem/sd-material-ui/blob/master/src/lib/components/AutoComplete.react.js) for details.